sqlite_query.c

```diff
@@ -44,52 +52,242 @@ sqlite_convert_to_pg(Oid pgtyp, int pgtypmod, sqlite3_stmt * stmt, int stmt_coli
 	Datum		value_datum = 0;
 	char	   *valstr = NULL;
 	int			affinity_for_pg_column = sqlite_affinity_eqv_to_pgtype(pgtyp);
-	int		 	value_byte_size_blob_or_utf8 = sqlite3_column_bytes(stmt, stmt_colid); // Compute always, void text and void BLOB will be special cases
-
-	if (affinity_for_pg_column != sqlite_value_affinity && sqlite_value_affinity == SQLITE3_TEXT)
-	{
-		sqlite_value_to_pg_error (pgtyp, pgtypmod, stmt, stmt_colid, sqlite_value_affinity, affinity_for_pg_column, value_byte_size_blob_or_utf8);
-	}
+				/* Compute always, void text and void BLOB will be special cases */
+	int		 	value_byte_size_blob_or_utf8 = sqlite3_column_bytes(stmt, stmt_colid);
 
 	switch (pgtyp)
 	{
+		case BOOLOID:
+			{
+				switch (sqlite_value_affinity)
+				{
+					case SQLITE_INTEGER: /* <-- proper and recommended SQLite affinity of value for pgtyp */
+						{
+							int			value = sqlite3_column_int(stmt, stmt_colid);
+							return (struct NullableDatum){BoolGetDatum(value), false};
+						}
+					case SQLITE_FLOAT:
+					case SQLITE_BLOB:
+						{
+							sqlite_value_to_pg_error (pgtyp, pgtypmod, stmt, attnum, sqlite_value_affinity, affinity_for_pg_column, value_byte_size_blob_or_utf8);
+							break;
+						}
+					case SQLITE3_TEXT:
+					{
+						if (value_byte_size_blob_or_utf8)
+							sqlite_value_to_pg_error (pgtyp, pgtypmod, stmt, attnum, sqlite_value_affinity, affinity_for_pg_column, value_byte_size_blob_or_utf8);
```

Comment:
I think the BOOLEAN text can be converted to postgres BOOL easily. Could you explain why this error happen?

Reply:
Yes. We can read, but this will not consistent. Data converting in `sqlite_query.c` is not all. Now there is no adopted `WHERE` predicate for ISO:SQL behavior, `text` affinity `bool`s will invisible for `WHERE`. Please read opinion of @t-kataym and my message before it. The short answer is this is `mixed affinity` case like UUID and we need special SQLite side data normalization like `unhex`, but for SQLite boolean data with `text` affinity. Maybe _mixed affinity boolean_ will my PR in a future.

Reply:
Resolved.